src/dbnode/storage/index/block.go

```diff
@@ -894,6 +895,10 @@ func (b *block) Aggregate(
 	}
 
 	aggOpts := results.AggregateResultsOptions()
+	// ensure we iterate the provided filter fields in order.
+	sort.Slice(aggOpts.FieldFilter, func(i, j int) bool {
```

Comment:
Will this cause a data race since `block.Aggregate(...)` happens in parallel across blocks?

Reply:
good catch, thanks.

Reply:
addressed by moving the sort up into `index.go:Aggregate()` before we call into the blocks.